core/src/main/java/de/medavis/lct/core/patcher/BomPatcher.java

```diff
@@ -0,0 +1,256 @@
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package de.medavis.lct.core.patcher;
+
+import com.google.common.io.ByteStreams;
+
+import de.medavis.lct.core.Configuration;
+
+import org.apache.commons.lang3.StringUtils;
+import org.cyclonedx.Version;
+import org.cyclonedx.exception.ParseException;
+import org.cyclonedx.generators.BomGeneratorFactory;
+import org.cyclonedx.model.Bom;
+import org.cyclonedx.model.Component;
+import org.cyclonedx.model.License;
+import org.cyclonedx.model.LicenseChoice;
+import org.cyclonedx.parsers.BomParserFactory;
+import org.jetbrains.annotations.NotNull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.URI;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+public class BomPatcher {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(BomPatcher.class);
+
+    private static final Pattern EXPRESSION_OR_PATTERN = Pattern.compile("^\\((?<id1>.*) OR (?<id2>.*)\\)$");
+
+    private final SpdxLicenseManager spdxLicenseManager;
+    private final ComponentMetaDataManager componentMetaDataManager;
+
+    private final Configuration configuration;
+
+    public BomPatcher(@NotNull Configuration configuration) {
+        this.configuration = configuration;
+
+        spdxLicenseManager = SpdxLicenseManager.create();
+        componentMetaDataManager = new ComponentMetaDataManager();
+    }
+
+    private void init() {
+        configuration
+                .getSpdxLicensesUrl()
+                .ifPresentOrElse(
+                        url -> spdxLicenseManager.load(URI.create(url.toString())),
+                        spdxLicenseManager::loadDefaults
+                );
+
+        configuration
+                .getComponentMetadataUrl()
+                .map(url -> URI.create(url.toString()))
+                .ifPresent(componentMetaDataManager::load);
+
+        componentMetaDataManager.validateLicenseMappedNames(spdxLicenseManager.getSupportedLicenseNames());
```

Comment:
What's the point in validating the licenses if you don't care about the result?

Reply:
Renamed and fixed